USAGE.md

```diff
@@ -0,0 +1,39 @@
+# Usage
+   
+
+## Drone Server Setup
+
+_**If you have a PX4/MAVSDK server instance running on an external device the below steps are not necessary.**_
+
+### Install PX4 Toolchain (Ubuntu/WSL only)
+
+Next we download the PX4 source code and use the normal Ubuntu installer script to set up the developer environment. This will install the toolchain for Gazebo simulation, JMAVSim simulation and Pixhawk/NuttX hardware.
+
+To install the development toolchain:
+1. Open your shell environment (`bash`, `zsh`, WSL shell)
+2. Run the installation script.
```

Comment:
This isnt nessecarily unless the `make build-all` command failed for whatever reason. They should then use the internal version. 

Probably best if we change the submodule folder to match the PX4-Autopilot folder name and enforce it being in the same spot. Then we can route the easy commands through the Makefile?

Reply:
Well in that case, `make build-all` would fail for macOS since it is not ubuntu (no `apt`). 

I guess I really didn't realize what the submodules were. Why do we need to clone/`make` these submodules when they are not necessary for the package to be built and installed? I mean sure, they _can_ be needed, but just because you want to build the package from source, doesn't necessarily mean you need the PX4 toolchain downloaded. It could already be running on another machine.

I feel like the same applies to the other submodules. Yeah, they are good to have, but we don't need to build all those things just to install the package. The package isn't really dependent on them being built.

Reply:
We can make from source with `make install` for just the python stuff

Reply:
`make build-all` would be dependent on Ubuntu 18.04/20.04 anyways since it calls ubuntu.sh, we could flush this out more, but we use the same ubuntu.sh script from px4 for the Windows/WSL stuff anyways, so...

Reply:
I agree that I can replace this stuff with the Makefile commands, that comment turned more into a comment on `make build-all`. Remember, I can still test and run pteranodon on a mac, I just have to use the docker container instead.

Reply:
(The above comment happened before GitHub showed yours)

Reply:
Why not just `make build` for the essentials and `make build-all` for everything?

Reply:
Yeah add a `make build` for just PX4 and leave the other stuff for just make build-all? I am fine with that, may need to adjust how I setup the make chaining stuff a little bit or just call scripts located in the third-party folder or elsewhere

Reply:
I made that to kind of lightly enforce PX4 being in a common location for script development purposes

Reply:
I also made that cause I needed to reinstall everything on one of my computers and it worked for Ubuntu so I left it

Reply:
I was thinking `make build` for the pteranodon +  install (last two steps of `make build-all`) and `make build-all` would stay the same. 

That way it is saying, `build` builds the project, `build-all` builds everything and the project